### DISCUSSION TITLE
Gizmos and sprites: How can I draw a SpriteBundle on top of a Gizmo?

### DISCUSSION THREAD

**Discussion by ryfactor** [bevyengine/bevy#11601] [Q&A] [answered]
I would like to be able to draw some gizmos above sprites, and some gizmos below sprites. Can I achieve this currently, or does this feature still need to be built? If it's possible, how can best I implement it? Thanks for making such an amazing and featureful engine by the way :)

**Comment by tim-blackbird** [comment]
This isn't currently possible due to how 2D rendering works and that gizmos are all drawn together as a single mesh as an optimization

**Reply by nicopap** [reply]
I guess it's now technically possible with gizmo configs. Since gizmo configs create individual meshes per config.

**Reply by tim-blackbird** [reply]
It could be. The 2D gizmo render sorting is currently hardcoded to be after everything else
https://github.com/bevyengine/bevy/blob/fb124c35beb96c7374764aa51517a12a3183c992/crates/bevy_gizmos/src/pipeline_2d.rs#L184

**Comment by ryfactor** [comment]
Thanks for your answers so far, and sorry for taking so long to get back to you all.
I had experimented a little with changing the depth bias of a gizmo when used in conjunction with a bevy sprite, and the changes in my fork (https://github.com/bevyengine/bevy/compare/main...ryfactor:bevy:gizmo-layers) enabled me to crudely change the layer of a gizmo, with the following results in pictures attached.

Now I don't know if this is an acceptable way to do this in bevy roadmap or performance terms, and I don't want to cause havoc by passing this off as a good way to do this and attempt to have it merged. However, I would like some feedback on whether this approach is heading in the right direction, and what tweaks might be needed to enable changing the depth bias of a gizmo in a truly guru way?

If you run the example, you can push the round gizmos forward and backward through the z-axis. Push too far and they lose visibility in the plane (which I thought was unusual). Also, there's no good way I could see to re-set the `sort_key` to infinity by changing the depth bias to a large negative number to be able to snap a gizmo to the front. It's like the `sort_key` is only set on initialization or something, so I don't know how to mutate that dynamically. Some gentle advice would be supremely appreciated. Love xx

![1_round gizmos in front](https://github.com/bevyengine/bevy/assets/6005108/cfa7e662-8da4-483e-b6a9-1c9acdb8265c)
![2_round gizmos behind straight](https://github.com/bevyengine/bevy/assets/6005108/01862051-7dfb-445b-8df5-1c011407236b)
![3_round gizmos behind sprite layer](https://github.com/bevyengine/bevy/assets/6005108/43416f4c-58ef-4076-b237-e2f4242c6021)

**Comment by gloridifice** [comment, marked as answer]
I encountered the same problem and opened an issue #13027 . [jgayfer](https://github.com/jgayfer) offered a solution using RenderLayers and it worked successfully. Hope that can help you.

**Comment by ryfactor** [comment]
Wow, thank you very much @gloridifice for linking to the RenderLayers solution, the very illustrative example in the issue is very helpful.
